_posts/2023-08-15-securing-wildfly-apps-saml-openshift.adoc

```diff
@@ -0,0 +1,380 @@
+
+6. From the *Role Mapping* tab, assign *alice* the *user* role. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#proc-assigning-role-mappings_server_administration_guide[assign a role] to a user.
+
+7. Create a new client called *simple-webapp-saml*. Select *saml* as the *Client type* (or as the *Client Protocol*, depending on your Keycloak version). For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#_client-saml-configuration[create a SAML client].
+
+Once the new client has been created, in the *Settings* tab, scroll down to the *SAML capabilities* section and
+set the *Name ID format* to *email*. Then set *Force name ID format* to *On*. Then click on *Save*.
+
+== Download the SAML Keys
+
+From your *simple-webapp-saml* client in the Keycloak Admin Console, click on the *Keys* tab.
+
+Click on the *Export* button in the *Signing keys config* to export the SAML keys to a keystore.
+
+Take note of the aliases and passwords that you specify here since these will be used
+when updating the `keycloak-saml.xml` file.
+
+We will set the *Key alias* to *simple-webapp-saml*, the *Key password* to *password*,
+the *Realm certificate alias* to *myrealm*, and the *Store password* to *password* and then
+click on *Export* to download the corresponding `keystore.jks` file.
+
+Create an OpenShift secret using this keystore by running the following command:
+
+[source]
+----
+oc create secret generic simple-webapp-saml-secret --from-file=/PATH/TO/keystore.jks
+----
+
+== Download and Edit the Keycloak Adapter Configuration File
+
+From your *simple-webapp-saml* client in the Keycloak Admin Console, click on the *Action* dropdown in the top right corner
+and select *Download Adapter Config*.
+
+For the *Format option*, select the *Keycloak SAML Adapter keycloak-saml.xml*, and download the file and place it in the example
+application's `WEB-INF` directory, i.e., place the `keycloak-saml.xml` file in
+`/PATH/TO/ELYTRON/EXAMPLES/simple-webapp-saml/src/main/webapp/WEB-INF`.
+
+Update the `keycloak-saml.xml` file as follows:
+
+1. Set the *SP* *entityID* to *"simple-webapp-saml"*
+2. Set the *SP* *logoutPage* to *"/simple-webapp-saml"*
+3. Replace the *SP* *Keys* configuration with the following configuration, being sure to use the aliases
+and passwords you specified when exporting the SAML keys to the `keystore.jks` file:
+
+[source]
+----
+<Keys>
+    <Key signing="true">
+        <KeyStore password="password" file="/etc/keycloak-saml-secret-volume/keystore.jks">
+            <PrivateKey alias="simple-webapp-saml" password="password"/>
+            <Certificate alias="myrealm"/>
+        </KeyStore>
+    </Key>
+</Keys>
+----
+
+Push this new file to the `simple-webapp-saml` directory in your `elytron-examples` fork:
```

Comment:
I think we should mention that the files should be available on the default GH branch of the fork. My fork still had "master" as the default branch, therefore  "helm install" failed with the status "Failed (InvalidContextDirectory)." This was resolved after pushing the changes to "master".

Reply:
Added.